test_patches_distortion using == in float comparison

### Description
`visualization.wcsaxes` is using some equalities in float array comparisons in
https://github.com/astropy/astropy/blob/20d6808fa5e00d18df518a9e93ca03fcad092fd3/astropy/visualization/wcsaxes/tests/test_images.py#L690-L692

### Expected behavior
Test should pass on all supported architectures.

### Actual behavior
Second assertion fails on i5 Mac, where
```python
>>> r2.get_xy()[0] - [266.4 , -29.25]
array([0.00000000e+00, 7.10542736e-15])
```

### Steps to Reproduce
With both WCSLIB 7.6 and 7.7 from #12034
```python
astropy.test(package='visualization', args='-vk test_patches_distortion', remote_data=True)
...
>       assert (r2.get_xy()[0] == [266.4, -29.25]).all()
E       assert False
E        +  where False = <built-in method all of numpy.ndarray object at 0x12fa51f30>()
E        +    where <built-in method all of numpy.ndarray object at 0x12fa51f30> = array([266.4 , -29.25]) == [266.4, -29.25]
E             Use -v to get the full diff.all

../lib/python3.9/site-packages/astropy/visualization/wcsaxes/tests/test_images.py:692: AssertionError
```
About all in `test_images` is marked `remote_data` and thus skipped on macOS as well as 32bit, although I don't see any use of `IMAGE_REFERENCE_DIR` or other remote data in this particular test, and it passes in fact without network connection after deleting the decorators.

### System Details
Running tests with Astropy version 5.0.dev565+g60ea08e52.d20210810.
Running tests in lib/python3.9/site-packages/astropy/visualization.
Date: 2021-08-10T15:48:40
Platform: macOS-10.14.6-x86_64-i386-64bit
Executable: /opt/bin/python3.9
Full Python Version: 
3.9.6 (default, Jul 19 2021, 15:17:01) 
[Clang 11.0.0 (clang-1100.0.33.17)]
encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.21.1
Scipy: 1.7.1
Matplotlib: 3.5.0.dev1539+gf8f693922c.d20210728

--- a/astropy/coordinates/tests/test_angular_separation.py
+++ b/astropy/coordinates/tests/test_angular_separation.py
@@ -71,7 +71,7 @@ def test_proj_separations():
     # returns an Angle object
     assert isinstance(sep, Angle)
 
-    assert sep.degree == 1
+    assert_allclose(sep.degree, 1.)
     assert_allclose(sep.arcminute, 60.)
 
     # these operations have ambiguous interpretations for points on a sphere

--- a/astropy/coordinates/tests/test_api_ape5.py
+++ b/astropy/coordinates/tests/test_api_ape5.py
@@ -210,7 +210,7 @@ def test_frame_api():
     coo1 = ICRS(ra=0*u.hour, dec=0*u.deg)
     coo2 = ICRS(ra=0*u.hour, dec=1*u.deg)
     # `separation` is the on-sky separation
-    assert coo1.separation(coo2).degree == 1.0
+    assert_allclose(coo1.separation(coo2).degree, 1.0)
 
     # while `separation_3d` includes the 3D distance information
     coo3 = ICRS(ra=0*u.hour, dec=0*u.deg, distance=1*u.kpc)

--- a/astropy/coordinates/tests/test_frames.py
+++ b/astropy/coordinates/tests/test_frames.py
@@ -663,7 +663,7 @@ def test_sep():
     i2 = ICRS(ra=0*u.deg, dec=2*u.deg)
 
     sep = i1.separation(i2)
-    assert sep.deg == 1
+    assert_allclose(sep.deg, 1.)
 
     i3 = ICRS(ra=[1, 2]*u.deg, dec=[3, 4]*u.deg, distance=[5, 6]*u.kpc)
     i4 = ICRS(ra=[1, 2]*u.deg, dec=[3, 4]*u.deg, distance=[4, 5]*u.kpc)

--- a/astropy/coordinates/tests/test_representation.py
+++ b/astropy/coordinates/tests/test_representation.py
@@ -2066,7 +2066,7 @@ def test_unitphysics(unitphysics):
 
     asphys = obj.represent_as(PhysicsSphericalRepresentation)
     assert asphys.phi == obj.phi
-    assert asphys.theta == obj.theta
+    assert_allclose(asphys.theta, obj.theta)
     assert_allclose_quantity(asphys.r, 1*u.dimensionless_unscaled)
 
     assph = obj.represent_as(SphericalRepresentation)

--- a/astropy/visualization/wcsaxes/tests/test_images.py
+++ b/astropy/visualization/wcsaxes/tests/test_images.py
@@ -687,7 +687,7 @@ class TestBasic(BaseImageTests):
         # the input(center) is a tuple or a SkyCoord object.
         assert (r1.get_xy() == r2.get_xy()).all()
         assert np.allclose(r1.get_xy(), r3.get_xy())
-        assert (r2.get_xy()[0] == [266.4, -29.25]).all()
+        assert np.allclose(r2.get_xy()[0], [266.4, -29.25])
 
         return fig
 

